typed/remove_test.go

```diff
@@ -0,0 +1,457 @@
+package typed
+
+import (
+	"fmt"
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/v3/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v3/value"
+)
+
+var (
+	// Short names for readable test cases.
+	_NS  = fieldpath.NewSet
+	_P   = fieldpath.MakePathOrDie
+	_KBF = fieldpath.KeyByFields
+	_V   = value.NewValueInterface
+	_I   = fieldpath.NewIndexPathElement
+)
+
+func TestRemoveDeduced(t *testing.T) {
+	var cases = []struct {
+		object          YAMLObject
+		remove          *fieldpath.Set
+		expect          YAMLObject
+		expectItemsOnly YAMLObject
```

Comment:
I like that you're testing both at the same, that's so much simpler :-)

Reply:
BUT, for CRDs everything is treated as a map, so everything is an item, so I can't see any test where the two are not the same?

Reply:
Right. Previously there was the "skip atomic" part. Would that make it compatible with crds?